addons/token-exchange/s3_secret_handler.go

```diff
@@ -89,16 +89,12 @@ func (s s3SecretHandler) syncBlueSecret(name string, namespace string, c *blueSe
 
 	var storageClusterRef *v1alpha1.StorageClusterRef
 	for _, mirrorPeer := range mirrorPeers {
-		storageClusterRef = common.GetCurrentStorageClusterRef(&mirrorPeer, c.clusterName)
-		if storageClusterRef != nil {
-			break
+		storageClusterRef, err = common.GetCurrentStorageClusterRef(&mirrorPeer, c.clusterName)
+		if err != nil {
+			return err
 		}
```

Comment:
why is the `break` not required anymore?

Reply:
Earlier it broke the flow here and returned [here](https://github.com/red-hat-storage/odf-multicluster-orchestrator/pull/62/files#diff-6d5da978d6ca2ab0b53059f5a528cc4369814fb54901dca598cb57439ab5a1aeL99) which looks like an overkill since if the error is not nil, then it makes sense to return it. `storageClusterRef` will be `nil` if `err` is not `nil`.

Reply:
I don't understand the purpose of this.

It keeps updating the same variable until the loop ends or there's no reference found and if it's the latter case, then it returns an error.

Shouldn't it return an error if no reference is found and break on the first match?

Reply:
Fixed.